pyvista/core/dataobject.py

```diff
@@ -751,6 +788,46 @@ def __getstate__(self):
 
     def __setstate__(self, state):
         """Support unpickle."""
+
+        def _is_pyvista_format(state_):
+            return isinstance(state_, dict) and 'vtk_serialized' in state_
+
+        def _is_vtk_format(state_):
+            # Note: The vtk state has format ( function, (dict,) )
+            return (
+                isinstance(state_, tuple)
+                and len(state_) == 2
+                and isinstance(state_[1], tuple)
+                and len(state_[1]) == 1
+                and isinstance(state_[1][0], dict)
+            )
+
+        if _is_pyvista_format(state):
+            self._unserialize_pyvista_pickle_format(state)
+        elif _is_vtk_format(state):
+            self._unserialize_vtk_pickle_format(state)
+        else:
+            raise RuntimeError(
+                f"Cannot unpickle '{self.__class__.__name__}'. Invalid pickle format."
+            )
+
+    def _unserialize_vtk_pickle_format(self, state):
+        """Support unpickle of VTK's format."""
+        # The vtk state has format: ( function, (dict,) )
+        state_dict = state[1][0]
+        self.__dict__.update(state_dict)
```

Comment:
This will set a "Serialized" attribute with the object's serialized string as the value. Instead, this should be popped from the dict after being unserialized so that this attribute is not set

Reply:
Instead of updating from the state dict directly, `self` is now updated from a separate `'_PYVISTA_STATE_DICT'`